### PR TITLE
fix: force HTTP1.1 for operator webhooks to remidiate CVE-2023-44487

### PR DESCRIPTION
**Pull request by johannes94**
## Description

Some other projects mainly Kubernetes claim that the [golang fix](https://github.com/golang/net/commit/b225e7ca6dde1ef5a5ae5ce922861bda011cfabd) for CVE-2023-44487 only fixes part of the problem and still leave room for DoS attacks.

https://github.com/kubernetes/kubernetes/pull/121120
https://github.com/kubernetes/kubernetes/issues/121197
https://github.com/golang/go/issues/63417#issuecomment-1758858612

Current remediation advice for (Red Hat) operators is to disable HTTP2 for webhooks and metrics server, if they use TLS.

This PR forces HTTP1.1 for the operator webhook server. Metrics server configuration remains untouched, since we're not using TLS.

## Checklist
- [x] Investigated and inspected CI test results
- ~~[ ] Unit test and regression tests added~~
- ~~[ ] Evaluated and added CHANGELOG entry if required~~
- ~~[ ] Determined and documented upgrade steps~~
- ~~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

CI is sufficient

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by vladbologa — APPROVED** [review]

**Comment by vladbologa** [comment]
Thanks, LGTM!

File: operator/main.go [hunk 1]
```diff
@@ -203,17 +208,37 @@ func run() error {
...
+		if enableHTTP2 {
+			return
+		}
+		c.NextProtos = []string{"http/1.1"}
+	}
+
 	opts := webhook.Options{
 		Port: 9443,
+		TLSOpts: []func(*tls.Config){
+			disableHTTP2,
```
[the remark this answers is not in the document]
Comment (johannes94) [reply operator/main.go:238]: TBH. I didn't put much though into that, just copied the solution from the remediation guideline.

[the remark this answers is not in the document]
Comment (mclasmeier) [reply operator/main.go]: Find it better to read this way!

[the remark this answers is not in the document]
Comment (johannes94) [reply operator/main.go]: @mclasmeier please take another look

**Pull request merged**